[Draft] Add GENERATE_ARRAY function

dttung2905 opened the pull request:
TODO list:
- [x] implement `transformToStringValuesMV` method
- [x] testing
- [x] Register the batch function


Instructions:
1. The PR has to be tagged with at least one of the following labels (*):
   1. `feature`
   2. `bugfix`
   3. `performance`
   4. `ui`
   5. `backward-incompat`
   6. `release-notes` (**)
2. Remove these instructions before publishing the PR.
 
(*) Other labels to consider:
- `testing`
- `dependencies`
- `docker`
- `kubernetes`
- `observability`
- `security`
- `code-style`
- `extension-point`
- `refactor`
- `cleanup`

(**) Use `release-notes` label for scenarios like:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed

xiangfu0 commented on pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java:
```diff
@@ -0,0 +1,223 @@
+
+  @Override
+  public int[][] transformToIntValuesMV(ValueBlock valueBlock) {
+    return new int[0][];
```
Here we should convert array literal and cache it.
Check how `ArrayLiteralTransformFunction` did here.

dttung2905 replied:
Yes. Its on my to do list too. No worries :smile:

xiangfu0 commented on pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayGenerateTransformFunction.java:
```diff
@@ -0,0 +1,223 @@
+
+  @Override
+  public int[] transformToDictIdsSV(ValueBlock valueBlock) {
+    return new UnsupportedOperationException();
```
throw new UnsupportedOperationException();

dttung2905 replied:
will fix it and the relevant functions as well. I was speeding through this yesterday sorry

dttung2905 commented:
Hi @xiangfu0 sorry I was really busy with works ( pre-Christmas period :sweat_smile: ). I just fixed the unit test for the failed test `testCalciteFunctionMapAllRegistered` by whitelisting it under `IGNORED_FUNCTION_NAMES`. The next time CI is run, the tests should pass.

Is there anything I'm mising before the review ? Do I need to 
- add the test for scalar function changes in `pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java`. I look up the other scalar function but can't find any test for those. cmiiw :thinking: 
- remove the changes in `pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java` as we have implemented the batch function already?
